language/move-vm/runtime/src/loader.rs

```diff
@@ -831,7 +827,7 @@ impl Loader {
     }
 
     // Expects all modules to be on chain. Gives an invariant violation if it is not found
-    fn load_module_expect_no_missing_dependencies(
+    fn load_module_expect_not_missing(
```

Comment:
For some reason I like the old name (i.e., the one with the `_no_missing_dependencies` part). It signals to me that not only the module bearing `id: &ModuleId` is not missing, but also all its dependencies, direct or transitive, are not missing too. But I don't have a strong preference here and I might be wrong with the understanding.

Reply:
The old naming was cool, but wrong :) 

The thing it is checking is that the actual module being loaded is missing or not, not its dependencies

Reply:
Yes, the function checks the actual module being loaded is missing or not, but also all the dependencies of this module as far as I understand the code.

Here is the convoluted call stack, but this is indeed a recursion.

```
load_module_verify_no_missing_dependencies(&ModuleId)
  load_module(&ModuleId)
    deserialize_and_verify_module(&ModuleId)
      verify_module_expect_no_missing_dependencies(&CompiledModule)
        verify_module(&CompiledModule)
          let immediate_deps: Vec<ModuleId> = (module: &CompiledModule).immediate_dependencies();
          load_dependencies_verify_no_missing_dependencies(Vec<ModuleId>)
            for module_id in immediate_deps { load_module_verify_no_missing_dependencies(&module_id) } --> the recursion here
```

Reply:
`verify_module_expect_no_missing_dependencies` is always called regardless of the flag. 
The only thing the flag is doing is is determining what to do if an error is encountered from `data_store.load_module(id)` 
Hence I switched the name of the function to based on the actual behavior that is switching, whether or not the specified module loads from storage or not

Reply:
Ah, now I understand the reason. Thank you! The new name is better :)